### DISCUSSION TITLE
Support of esp32 MCU

### DISCUSSION THREAD

**Discussion by ABBAGABBA** [MobiFlight/MobiFlight-Connector#1397] [Ideas]
Hello, everyone! 

My appologize if that request already is!
Please Add the support of esp32 MCU's to MobiFlight, i tried to do it myself but it is dificult for me.
Get support of esp32 MCU's it is important, because is cheaper them Mega boards and more productive.

**Comment by DocMoebiuz** [comment]
i converted this into a discussion.

Please other users who are interested (also in creating and maintaing it) chime in.

**Comment by elral** [comment]
@ABBAGABBA which board you are thinking about? There are a lot of different boards on the market which uses also different versions of the ESP32 (-S1 / -S2 / -S3). Then they also differ in the number of pins they provide. All of them are in the range of ~25 Pins, but please correct me if I am wrong and there is an availbale board with much more pins.
All these makes it not easy to decide for one board. And at MF already supports the Raspberry Pi Pico which is powerful and has 26 pins.

**Comment by ABBAGABBA** [comment]
@elral , i have esp32-wroom-32 ![IMG20231113220621](https://github.com/MobiFlight/MobiFlight-Connector/assets/93386330/87623c6c-6c9f-4e01-a530-a0751f4b1aec)
I agree with you about raspberry pico, but i think that if MB be support more different bord it will give more variants for building of cocpits, because not every one have opportunity buy for example raspberry pico. 

Thank you for your attention to this discussion!

**Comment by tigert** [comment]
The ESP32 S2 Mini seems an interesting board, it is very compact but has 18 analog inputs (they seem to be in two banks, so I don't know if there are some restrictions, need to study that) and it seems to work with the ESP32_S2_mini branch in the mobiflight-firmwaresource fork of @elral.

I built it with vscode / platformio and it flashed fine, and works in MobiFlight after putting the board.json file in place, maybe give it a try?

![WEMOS-ESP32-S2-MINI](https://github.com/MobiFlight/MobiFlight-Connector/assets/2587818/b18ad36a-a4d4-4655-bba1-e5b0c2561da7)

**Reply by tigert** [reply]
Builtin led is pin15 on this board.

**Reply by ABBAGABBA** [reply]
Thank you! Could you share board.json for esp32? I try to compare with my file which i tryed to create.

**Comment by JBFUK** [comment]
This is interesting, so ESP32 is supported?  Agree regards them being far superior to the Arduino boards.
